### ISSUE TITLE
Landscape mode

### ISSUE THREAD

**Issue by omarojo** [nateemma/FilterCam#1]
Have you figure out how to take video/photos in landscape mode with GOUimage2 ?  I dont see any parameters to enable landscape mode in gpuimage2

**Comment by nateemma** [comment]
Not exactly. The current code sort of works in landscape mode if you use the back camera. The aspect ratio is messed up though and I haven't had time to look at it. Landscape does not work with the front facing camera.

Actually, I may not have checked in that change since it was to the GPUImage code, I'll go look...

**Comment by nateemma** [comment]
OK, so I made the following changes to Camera.swift, in GPUImage2/Inputs:

1. In the declarations at the top of the class:

/*** iOS 8
    let cameraProcessingQueue = DispatchQueue.global(priority:DispatchQueue.GlobalQueuePriority.default)
    let audioProcessingQueue = DispatchQueue.global(priority:DispatchQueue.GlobalQueuePriority.default)
***/
    // post iOS 8:
    let cameraProcessingQueue = DispatchQueue.global(qos: DispatchQoS.QoSClass.default)
    let audioProcessingQueue = DispatchQueue.global(qos: DispatchQoS.QoSClass.default)

2. in the init routine

After:

        if (captureSession.canAddOutput(videoOutput)) {
            captureSession.addOutput(videoOutput)
        }
        captureSession.sessionPreset = sessionPreset


Add the following:

        // merged in changes to fix orientation issue with front camera
        var captureConnection: AVCaptureConnection!
        for connection in videoOutput.connections {
            for port in (connection as AnyObject).inputPorts {
                if (port as AnyObject).mediaType == AVMediaTypeVideo {
                    captureConnection = connection as? AVCaptureConnection
                }
            }
        }

        if captureConnection.isVideoOrientationSupported {        
            let location = self.location
            let videoOrientation = captureConnection.videoOrientation
            let deviceOrientation = UIDevice.current.orientation

            if ((location == .frontFacing) || (deviceOrientation == .portrait) || (deviceOrientation == .portraitUpsideDown) || (deviceOrientation == .unknown)){
                captureConnection.videoOrientation = .portrait
            } else if (location == .backFacing) {
                captureConnection.videoOrientation = (deviceOrientation == .landscapeLeft) ? .landscapeRight : .landscapeLeft
            }
       }

**Comment by omarojo** [comment]
oh ! :o this is very interesting. Thank you @nateemma will check this out to confirm.
